### ISSUE TITLE
Fix client validation in ClientCredentials-Example

### ISSUE THREAD

**Issue by x3a** [thephpleague/oauth2-server#1088]
After setting up the provided examples like described in the corresponding `README.md` the client validation for the ClientCredentials-Grant authorizes any client independent from which credentials were sent during the request.

**Example**

```
curl -s -X "POST" "http://localhost:4444/client_credentials.php/access_token" \ 
        -H "Content-Type: application/x-www-form-urlencoded" \
        -H "Accept: 1.0" \
        --data-urlencode "grant_type=client_credentials" \
        --data-urlencode "client_id=myawesomeapp" \
        --data-urlencode "client_secret=abc123" \
        --data-urlencode "scope=basic email" | jq .
                                   
{
  "token_type": "Bearer",
  "expires_in": 3600,
  "access_token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJhdWQiOiJteWF3ZXNvbWVhcHAiLCJqdGkiOiJkODBkMjQwNmVlNmFmMWFlN2I0OTU0MDU5MTNiNTc1ZjlkZDA0OTQxNzQ4OGMwOWQ0ODE5ZmRhNWYwYjg3NmUyMGE5NWI4ZTIxYjRiMWE3MSIsImlhdCI6MTU4MTMyNjM4NCwibmJmIjoxNTgxMzI2Mzg0LCJleHAiOjE1ODEzMjk5ODQsInN1YiI6IiIsInNjb3BlcyI6WyJiYXNpYyIsImVtYWlsIl19.S5sIXbOoUS1PytAMQ-sBfhT68od_EnbEfYBolq0Kw1i75To7x05gS1l41VilH_k6KFzPDWoM6hN-4r5sQjHQeYDClvc93AQs8iK7N5IXwlS0HPafKKSi5t23K1QlLPDDTJrdj5L-7d4qhfV4fD4cgGWZvoJHAnz86ikpe41YOV1U76UABLvdypc-Ie9KwXkfihYipc371zwagWdRp9_Uoay4q5w7gqJmdyE7BDjz-DtL3zlWPFqiP258dZmBb-5mjkgp2oRLe9vb1IrcdSOiKk8TpZ111Wua0uMqBe4IrgWKACBkU39-OJV8Nl0UEn9BBmJCgJ_royqn7g_itsodXA"
}
```
So this works as expected, but the next one doesn't.

```
curl -s -X "POST" "http://localhost:4444/client_credentials.php/access_token" \
        -H "Content-Type: application/x-www-form-urlencoded" \
        -H "Accept: 1.0" \
        --data-urlencode "grant_type=client_credentials" \
        --data-urlencode "client_id=nonexistent" \   
        --data-urlencode "client_secret=nonexistent" \
        --data-urlencode "scope=basic email" | jq .
                                                   
{
  "token_type": "Bearer",
  "expires_in": 3600,
  "access_token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJhdWQiOiJub25leGlzdGVudCIsImp0aSI6ImE1NTQ0ZWI2NjIxMjcxYWU2M2Y4MThmZTQxNzBkNjczNTA4Y2FlNDI0Y2UxNjU2MmJiYzA5ZTBiN2ViMDc1NmU3YTQwNGFjNTU2ZjQwN2Y2IiwiaWF0IjoxNTgxMzI2NzY0LCJuYmYiOjE1ODEzMjY3NjQsImV4cCI6MTU4MTMzMDM2NCwic3ViIjoiIiwic2NvcGVzIjpbImJhc2ljIiwiZW1haWwiXX0.Dy6ceoNbvNlUHDcccD1kUTkAmuTaPzDITC33rvy_SSHZ3IB1b7jPdTdVAexaWfGNxYiS5hYrtBPo2wKJlOsvw1GcibAMlNfR-4vS8GMTtqQ-nBpAYtZRm2CPMGUe_56PIuu_IklWvlq4zCxs-TeMJeNdF6zl854kD978o53YmMCFHUgxwQY__mDfqKK88q4zyt5KuaZLjtU6XpgXHVqebmALk1V9jmj1Wego7BMvcJSY-BRGwiU7uqAPAQ5_sJNQPoiqbZAXy46ChYFKUDYYcfGVX2PXPn4iZ3s6dEUoihE-vZz2smUch46F04YUyi0VYq6gacDfzl7KpS1uq_mZ4Q"
}
```
This should fail because of invalid credentials but doesn't. The problem lies in the provided `ClientRepository.php` within the examples, which contains a function `validateClient` that returns always `NULL`.

As stated in `ClientRepositoryInterface.php` from the oauth2 server this function should return a boolean, so I guess it should return `true` if the client validation has succeeded or `false` if it has failed.

**Comment by x3a** [comment]
I was able to fix this, this is how the correct response should look like:

```
curl -s -X "POST" "http://localhost:4444/client_credentials.php/access_token" \
        -H "Content-Type: application/x-www-form-urlencoded" \
        -H "Accept: 1.0" \
        --data-urlencode "grant_type=client_credentials" \
        --data-urlencode "client_id=nonexistent" \
        --data-urlencode "client_secret=nonexistent" \
        --data-urlencode "scope=basic email" | jq .
                  
{
  "error": "invalid_client",
  "error_description": "Client authentication failed",
  "message": "Client authentication failed"
}
```
See: [Pull Request](https://github.com/thephpleague/oauth2-server/pull/1089)

**Comment by Sephster** [comment]
Fixed in pull request #1089
